### PR TITLE
feat: list viewers in a drawer

### PR DESCRIPTION
**Pull request by chungwwei**
image

### CODE REVIEW

**Comment by kevmo314** [comment]
Nice! Can we hide the header if there are no users in that section? Also post this to the discord when it gets in :)

**Comment by KeN095** [comment]
Yup, it looks good! It might look better if icons appeared to the left of each header, but having just the header there is enough to portray the meaning.

**Review by KeN095 — APPROVED** [review]

**Review by juaoose — APPROVED** [review]
Should we remove the commented code?

File: lib/components/drawer/end_drawer.dart [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+class LeftDrawerWidgetState extends State<LeftDrawerWidget> {
+  Future<dynamic>? viewersFuture;
+
+  _viewersFuture(String channel) async {
+    Uri uri = Uri.parse('https://tmi.twitch.tv/group/user/$channel/chatters');
+    final res = await http.get(uri);
+    if (res.statusCode == 200) {
+      List<String> broadcasterList =
+          List<String>.from(jsonDecode(res.body)['chatters']['broadcaster']);
```
Comment (kevmo314) [inline lib/components/drawer/end_drawer.dart:25]: Store `jsonDecode(req.body)` in a var so it only has to be decoded once

File: lib/components/drawer/end_drawer.dart [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+  State<LeftDrawerWidget> createState() => LeftDrawerWidgetState();
+}
+
+class LeftDrawerWidgetState extends State<LeftDrawerWidget> {
+  Future<dynamic>? viewersFuture;
+
+  _viewersFuture(String channel) async {
+    Uri uri = Uri.parse('https://tmi.twitch.tv/group/user/$channel/chatters');
+    final res = await http.get(uri);
+    if (res.statusCode == 200) {
```
Comment (kevmo314) [inline lib/components/drawer/end_drawer.dart:23]: Use guard clauses instead, like put the error cases in the conditional. In other words,

```
if (res.statusCode != 200) {
  return {...}
}
...decode...
```

As a general rule, the last return should be the "success" case of the function.

**Pull request merged**